modules/netaddress.go

```diff
@@ -15,6 +20,9 @@ func (na NetAddress) Host() string {
 	if err != nil {
 		return ""
 	}
+	if na.IsValid() != nil {
+		return ""
+	}
```

Comment:
I added this check to discourage use of partial addresses. Like before how we did hostname discovery by initially setting the NetAddress to `localhost:port` and then later modifying only the hostname. This seemed to be the cause of a few bugs so I figured discouraging it was worth it.

If you think they shouldn't be there I'll remove them.